OpenOversight/app/main/views.py

```diff
@@ -416,6 +416,19 @@ def edit_department(department_id):
                 if rank:
                     new_ranks.append((rank, order))
                     order += 1
+            updated_ranks = form.jobs.data
+            if len(updated_ranks) < len(original_ranks):
+                deleted_ranks = [rank for rank in original_ranks if rank.job_title not in updated_ranks]
+                for rank in deleted_ranks:
+                    if Assignment.query.filter(Assignment.job_id.in_([rank.id for rank in deleted_ranks])).count() == 0:
+                        db.session.delete(rank)
+                        db.session.commit()
+                    else:
+                        formatted_rank = rank.job_title.replace(" ", "+")
+                        link = '/department/{}?name=&badge=&unique_internal_identifier=&rank={}&min_age=16&max_age=100&submit=Submit'.format(department_id, formatted_rank)
+                        flash(Markup('You attempted to delete a rank, {}, that is in use by <a href={}>the linked officers</a>.'.format(rank, link)))
+                    return redirect(url_for('main.edit_department', department_id=department_id))
```

Comment:
I believe this return statement being indented inside the for loop means that only one department can be edited or deleted at a time

Reply:
`test_admin_can_delete_rank_not_in_use` only deletes a single rank, so we wouldn't be catching this via unit tests, I my understanding of the code is correct

Reply:
Good catch.  I added functionality to be able to delete multiple ranks that are not in use.  I also added a test for this.

Reply:
Looks good, marking resolved